src/components/skybox.js

```diff
@@ -280,6 +288,20 @@ AFRAME.registerComponent("skybox", {
 
       this.el.object3D.matrixNeedsUpdate = true;
     }
+
+    this.updateEnvironmentMap();
+  },
+
+  updateEnvironmentMap() {
+    this.skyScene.add(this.sky);
+    this.cubeCamera.update(this.el.sceneEl.renderer, this.skyScene);
+    this.el.setObject3D("mesh", this.sky);
+
+    const environmentMapComponent = this.el.sceneEl.components["environment-map"];
+
+    if (environmentMapComponent) {
```

Comment:
I would just check this at the top and bail out of even rendering anything if there is no environment-map component.